src/main/java/org/prebid/server/bidder/gotthamads/GothamAdsBidder.java

```diff
@@ -0,0 +1,145 @@
+import org.prebid.server.util.HttpUtil;
+import org.prebid.server.util.ObjectUtil;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+public class GothamAdsBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, GothamAdsImpExt>> TYPE_REFERENCE = new TypeReference<>() {
+    };
+    private static final String ACCOUNT_ID_MACRO = "{{AccountId}}";
+    private static final String X_OPENRTB_VERSION = "2.5";
+    private static final Map<Integer, BidType> ALLOWED_IMP_TYPES = Map.of(
+            1, BidType.banner,
+            2, BidType.video,
+            4, BidType.xNative
+    );
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public GothamAdsBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final GothamAdsImpExt impExt;
+        final Imp imp = request.getImp().get(0);
+        try {
+            impExt = parseImpExt(imp);
+        } catch (PreBidException e) {
+            return Result.withError(BidderError.badInput(e.getMessage()));
+        }
+
+        final BidRequest bidRequest = request.toBuilder()
+                .imp(List.of(cleanUpExt(imp)))
+                .build();
+
+        final HttpRequest<BidRequest> httpRequest = HttpRequest.<BidRequest>builder()
+                .method(HttpMethod.POST)
+                .uri(resolveEndpoint(impExt.getAccountId()))
+                .headers(makeHeaders(request))
+                .body(mapper.encodeToBytes(bidRequest))
+                .payload(bidRequest)
+                .build();
+        return Result.withValue(httpRequest);
+    }
+
+    private static Imp cleanUpExt(Imp imp) {
+        return imp.toBuilder().ext(null).build();
+    }
+
+    private static MultiMap makeHeaders(BidRequest bidRequest) {
+        final Device device = bidRequest.getDevice();
+        final MultiMap headers = HttpUtil.headers();
+
+        headers.set(HttpUtil.X_OPENRTB_VERSION_HEADER, X_OPENRTB_VERSION);
+        HttpUtil.addHeaderIfValueIsNotEmpty(
+                headers,
+                HttpUtil.USER_AGENT_HEADER,
+                ObjectUtil.getIfNotNull(device, Device::getUa)
+        );
+        HttpUtil.addHeaderIfValueIsNotEmpty(
+                headers,
+                HttpUtil.X_FORWARDED_FOR_HEADER,
+                ObjectUtil.getIfNotNull(device, Device::getIpv6)
+        );
+        HttpUtil.addHeaderIfValueIsNotEmpty(
+                headers,
+                HttpUtil.X_FORWARDED_FOR_HEADER,
+                ObjectUtil.getIfNotNull(device, Device::getIp)
+        );
+
+        return headers;
+    }
+
+    private GothamAdsImpExt parseImpExt(Imp imp) {
```

Comment:
Methods are still not in call order, as `parseImpExt` and `cleanUpExt` are called before `makeHeaders`, but defined after them.

Reply:
sure